x-pack/dockerlogbeat/pipelinemock/pipelines.go

```diff
@@ -85,3 +85,10 @@ func (pc *MockPipelineConnector) ConnectWith(beat.ClientConfig) (beat.Client, er
 
 	return c, nil
 }
+
+func (pc *MockPipelineConnector) GetConnectedClients() []*MockBeatClient {
+	pc.mtx.Lock()
+	defer pc.mtx.Unlock()
+
+	return pc.clients
+}
```

Comment:
ok although I am not sure why treating them as mutable or immutable isn't something that can come from the developer I like the proposal of the `HasConnectedClients` func

Reply:
A mutex is guarding access to `pc.clients`. If we are returning a slice, we are making the underlying memory mutable by an external caller which escapes the scope of the mutex. It's good practice not to do this, even (and I would argue especially) in test code. This can lead to very nasty bugs that are nearly impossible to diagnose. A caller may not be aware of this and start mutating the slice, not knowing that they are also muting what's in `MockPipelineConnector`.

Reply:
@taylor-swanson ty for the explanation; in my book it a better practice not to do redundant allocations just to treat something as immutable, now I understand that this is a golang idiocracy and maybe Rust does a better better job here, but even hey no need to worry about this argument as this is no longer the case we switched to your proposed function which made sense